old-branches.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/bash
+
+timerange=$1
+
+out=''
+branches=()
+while read -r line; do
+  elems=($line)
+  name=${elems[1]#"refs/heads/"}
+  if [[ ${name} == release* ]]; then
+    continue
+  fi
+  hash=${elems[0]}
+  date=$(git show -s --format="%cs" ${hash})
+  if [[ ${date} != ${timerange}* ]]; then
+    continue
+  fi
+  out+=$(printf "%s:%s" ${date} ${name})
+  out+="\n"
+  branches+=(${name})
+done <<< $(git ls-remote -h origin)
```

Comment:
A bit of an odd way to express this, since it takes the output of a command, turns it into a param, and then turns that param into the input of another command. I might do it like:

```
git ls-remote -h origin | while read -r line ; do 
   ...
done
```

(No action required, just sharing bash nerdery)

Reply:
This doesn't actually work: it collapses the whole output into a single line (with embedded `\n`s in it) and reads the whole thing all at once, only executing the loop body once. The method I settled on here was the simplest way I found to ensure it got properly broken up by newlines (without parsing/regexing for them/splitting on them myself).

Reply:
Weird! This works as expected for me:

```
git ls-remote -h origin | while read -r line ; do echo "zz - $line" ; done
```